### ISSUE TITLE
[Web] - Broken TextField in Semantics mode when using ListView and Scaffold’s AppBar on mobile browsers

### ISSUE THREAD

**Issue by rkunboxed** [flutter/flutter#154741]
### Steps to reproduce

1. Run a web app with ensureSemantics enabled on a mobile device
2. Render a TextField or two in a ListView inside a Scaffold with an AppBar
3. Attempt to tap on a TextField

### Expected results

The mobile keyboard should come up and stay up

### Actual results

The mobile keyboard comes up and closes again very quickly making it impossible to enter text.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
  WidgetsFlutterBinding.ensureInitialized().ensureSemantics();
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(title),
      ),
      body: Center(
        child: Padding(
          padding: const EdgeInsets.only(top: 40),
          child: SizedBox(
            width: 300,
            //The ListView can be substituted for a SingleChildScrollView with a Column and the issue still persists.
            child: ListView(
              children: [
                const TextField(
                  decoration: InputDecoration(
                    hintText: 'Prefix',
                    suffixIcon: Icon(Icons.abc),
                  ),
                ),
                const SizedBox(height: 30),
                const TextField(
                  decoration: InputDecoration(
                    hintText: 'First Name',
                    suffixIcon: Icon(Icons.abc),
                  ),
                ),
                const SizedBox(height: 30),
                const TextField(
                  decoration: InputDecoration(
                    hintText: 'Last Name',
                    suffixIcon: Icon(Icons.abc),
                  ),
                ),
                const SizedBox(height: 30),
                FilledButton(
                  onPressed: () {},
                  child: const Text('Submit'),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.24.1, on macOS 14.6.1 23G93 darwin-arm64, locale en-US)
    • Flutter version 3.24.1 on channel stable at /Users/rona/Sites/Tools/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5874a72aa4 (2 weeks ago), 2024-08-20 16:46:00 -0500
    • Engine revision c9b9d5780d
    • Dart version 3.5.1
    • DevTools version 2.37.2

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/to/macos-android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[!] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    ✗ Unable to get list of installed Simulator runtimes.
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/to/macos-android-setup for detailed instructions).

[✓] VS Code (version 1.92.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.94.0

[✓] Connected device (3 available)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 14.6.1 23G93 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 14.6.1 23G93 darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome 128.0.6613.120

[✓] Network resources
    • All expected network resources are available.
```

</details>

**Comment by rkunboxed** [comment]
A few other things to note:
Make the `ListView` a `Column` and there is no issue. Remove the `AppBa`r and there is no issue. Remove `ensureSemantics` and there is no issue.

This could be related to https://github.com/flutter/flutter/issues/141975 but error text is not a factor here.

**Comment by darshankawar** [comment]
Thanks for the detailed report. I was able to replicate this upon running on mobile browsers such as (ios safari, chrome and on Android).

https://github.com/user-attachments/assets/7af76027-271e-4377-8d85-e6c53c09e7f0

```
Stable : 3.24.2
Master: 3.25.0-1.0.pre.269
```

**Comment by mdebbar** [comment]
Few notes from a quick look:
1. This doesn't seem to be caused by the same root cause of https://github.com/flutter/flutter/issues/141975.
2. I wasn't able to repro this. The keyboard always shows up and stays up for me.

**Comment by Amansrivastava4590** [comment]
I also has the same problem when i am initialising ensureSemantics , keyboard is comes up and closing again that cause the keyboard rendering  in mobile web browser, is it any solution for this. @rkunboxed

**Comment by GODOM018** [comment]
I am able to reproduce this issue using **Flutter version 3.27.1.** Upon clicking "done" for the first time, the keyboard begins to close immediately when focusing on the text field. However, if I remove the `ensureSemantics`, the application behaves seamlessly.

### Minimal reproducible app

```
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();
  SemanticsBinding.instance.ensureSemantics();

  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('Keyboard error'),
        ),
        body: SingleChildScrollView(
          controller: ScrollController(),
          physics: ClampingScrollPhysics(),
          child: Column(
            children: [
              Container(
                color: Colors.blue,
                height: 200.0,
                width: double.infinity,
              ),
              TextFormField(
                decoration: InputDecoration(
                  label: Text('Insert a text'),
                  suffixIcon: Icon(Icons.abc),
                ),
              ),
              const SizedBox(height: 80.0),
            ],
          ),
        ),
      ),
    );
  }
}

```

### Local Flutter version 

Flutter 3.27.1 • channel stable •
https://github.com/flutter/flutter.git
Framework • revision 17025dd882 (4 weeks ago) • 2024-12-17
03:23:09 +0900
Engine • revision cb4b5fff73
Tools • Dart 3.6.0 • DevTools 2.40.2
